gridcomps/cap3g/CapGridComp.F90

```diff
@@ -51,15 +44,23 @@ subroutine setServices(gridcomp, rc)
       ! Attach private state
       _SET_NAMED_PRIVATE_STATE(gridcomp, CapGridComp, PRIVATE_STATE, cap)
 
-      ! Get Names of children
-      call MAPL_GridCompGet(gridcomp, hconfig=hconfig, _RC)
-      hconfig_params = HConfigParams(hconfig, 'extdata_name') 
-      call MAPL_HConfigGet(hconfig_params, value=cap%extdata_name, default='EXTDATA', _RC)
-      hconfig_params%label = 'history_name'
-      call MAPL_HConfigGet(hconfig_params, value=cap%history_name, default='HIST', _RC)
-      hconfig_params%label = 'root_name'
-      call MAPL_HConfigGet(hconfig_params, value=cap%root_name, _RC)
+      ! Disable extdata or history
+      call MAPL_ResourceGet(gridcomp, keystring='run_extdata', value=cap%run_extdata, default=.true., _RC)
+      call MAPL_ResourceGet(gridcomp, keystring='run_history', value=cap%run_history, default=.true., _RC)
 
+      ! Get Names of children
+      call MAPL_ResourceGet(gridcomp, keystring='extdata_name', value=cap%extdata_name, default='EXTDATA', _RC)
+      call MAPL_ResourceGet(gridcomp, keystring='root_name', value=cap%root_name, _RC)
+      call MAPL_ResourceGet(gridcomp, keystring='history_name', value=cap%history_name, default='HIST', _RC)
+
+      if (cap%run_extdata) then 
+         call MAPL_ConnectAll(gridcomp, src_comp=cap%extdata_name, dst_comp=cap%root_name, _RC)
+      end if
+      if (cap%run_history) then
+         !call MAPL_ConnectAll(gridcomp, src_comp=cap%root_name, dst_comp=cap%history_name, _RC)
+         outer_meta => get_outer_meta_from_inner_gc(gridcomp,_RC)
+         call outer_meta%connect_all(cap%root_name, cap%history_name, _RC)
```

Comment:
```suggestion
         call MAPL_ConnectAll(gridcomp, src_comp=cap%root_name, dst_comp=cap%history_name, _RC)
```
Test again this way to see if it now works.

Reply:
@tclune, are you changed this. I still get that error?

Reply:
ok - I should be able to troubleshoot once this is merged.